libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1890,6 +1890,27 @@ void TypeChecker::endVisit(BinaryOperation const& _operation)
 	_operation.annotation().isLValue = false;
 	_operation.annotation().isConstant = false;
 
+	if (_operation.getOperator() == Token::Equal || _operation.getOperator() == Token::NotEqual)
+	{
+		auto leftFunction = dynamic_cast<FunctionType const*>(leftType);
+		auto rightFunction = dynamic_cast<FunctionType const*>(rightType);
```

Comment:
```suggestion
		auto const* leftFunction = dynamic_cast<FunctionType const*>(leftType);
		auto const* rightFunction = dynamic_cast<FunctionType const*>(rightType);
```

Reply:
Why, it will deduce it to `Function type const*` regardless? Having it be `auto const*` would be redundant since the type is already explicitly stated in the `dynamic_cast`.

Reply:
I mean, that's our current style that we use everywhere.

For `auto const*` it does not matter, but I guess we do it for consistency with `auto const&` where it does matter (you get a copy if you declare the variable with plain `auto`). I think it's a common style rule to just always include the qualifiers with `auto`.